rskj-core/src/main/java/co/rsk/core/ReversibleTransactionExecutor.java

```diff
@@ -28,63 +28,107 @@
-                                                         ReceiptStore receiptStore,
-                                                         ProgramInvokeFactory programInvokeFactory,
-                                                         Block executionBlock,
-                                                         RskAddress coinbase,
-                                                         byte[] gasPrice,
-                                                         byte[] gasLimit,
-                                                         byte[] toAddress,
-                                                         byte[] value,
-                                                         byte[] data,
-                                                         byte[] fromAddress) {
-        Repository repository = track.getSnapshotTo(executionBlock.getStateRoot()).startTracking();
+/**
+ * Encapsulates the logic to execute a transaction in an
+ * isolated environment (e.g. no persistent state changes).
+ */
+@Component
+public class ReversibleTransactionExecutor {
 
-        byte[] nonce = repository.getNonce(new RskAddress(fromAddress)).toByteArray();
-        UnsignedTransaction tx = new UnsignedTransaction(nonce, gasPrice, gasLimit, toAddress, value, data, fromAddress);
+    private final RskSystemProperties config;
+    private final Repository track;
+    private final BlockStore blockStore;
+    private final ReceiptStore receiptStore;
+    private final ProgramInvokeFactory programInvokeFactory;
 
-        ReversibleTransactionExecutor executor = new ReversibleTransactionExecutor(config, tx, coinbase, repository, blockStore, receiptStore, programInvokeFactory, executionBlock);
-        return executor.executeTransaction();
+    @Autowired
+    public ReversibleTransactionExecutor(
+            RskSystemProperties config,
+            Repository track,
+            BlockStore blockStore,
+            ReceiptStore receiptStore,
+            ProgramInvokeFactory programInvokeFactory) {
+        this.config = config;
+        this.track = track;
+        this.blockStore = blockStore;
+        this.receiptStore = receiptStore;
+        this.programInvokeFactory = programInvokeFactory;
     }
 
-    public static TransactionExecutor executeTransaction(RskSystemProperties config,
-                                                         Repository track,
-                                                         BlockStore blockStore,
-                                                         ReceiptStore receiptStore,
-                                                         ProgramInvokeFactory programInvokeFactory,
-                                                         Block executionBlock,
-                                                         RskAddress coinbase,
-                                                         Web3.CallArguments args) {
+    public ProgramResult executeTransaction(
```

Comment:
I would remove this method from here and unwrap params into EthModule

Reply:
Done. I ended up moving `callConstant` to `EthModule` too:

* https://github.com/rsksmart/rskj/pull/489/commits/da77fb31684ba28b3d07fa8e5c1bf58548d7a3f0
* https://github.com/rsksmart/rskj/pull/489/commits/e1e0a40121eb30e99d47df989ec1bdeb62520e04